pkg/pillar/cmd/monitor/ipc_server.go

```diff
@@ -169,14 +170,24 @@ func (s *monitorIPCServer) sendIpcMessage(t string, msg any) error {
 	defer s.Unlock()
 
 	var err error
+	var data []byte
 
-	if data, err := json.Marshal(msg); err == nil {
+	if data, err = json.Marshal(msg); err == nil {
 		ipcMessage := ipcMessage{Type: t, Message: json.RawMessage(data)}
 		if data, err = json.Marshal(ipcMessage); err == nil {
-			log.Noticef("Sending IPC message: %s", string(data))
+			if t == "TpmLogs" {
+				log.Noticef("Sending IPC message: %s", t)
+			} else {
+				log.Noticef("Sending IPC message: %s", string(data))
+			}
 			_, err = s.codec.Write(data)
+
+			if err != nil {
+				log.Errorf("Failed to send IPC message: %v", err)
```

Comment:
I think it's better to  return the error and let the caller function decide whether to log it or to propagate it further up the stack

```suggestion
				return fmt.Errorf("Failed to send IPC message: %v", err)
```

Reply:
@europaul the function is called from 10 different places. I do not want to duplicate code

Reply:
I agree it might make sense in some places. I just want to avoiding producing excessive logs like due to logging the error every time we encounter it as well as propagating the same error up the stack
```
ERROR something not working!
ERROR function bla existed with error: something not working!
```